internal/cli/performance_advisor_opts.go

```diff
@@ -38,14 +38,28 @@ func (opts *PerformanceAdvisorOpts) ValidateProcessName() error {
 	return nil
 }
 
-// MarkProcessNameHostIDRequired marks processName or hostId as required in accordance with the service
+// MarkRequiredFlagsByService marks processName or hostId as required in accordance with the service
+//
 // Atlas: processName is required
+//
 // OM/CM: hostId is required
-func (opts *PerformanceAdvisorOpts) MarkProcessNameHostIDRequired(cmd *cobra.Command) func() error {
+func (opts *PerformanceAdvisorOpts) MarkRequiredFlagsByService(cmd *cobra.Command) func() error {
 	return func() error {
 		if config.Service() == config.CloudService {
 			return cmd.MarkFlagRequired(flag.ProcessName)
 		}
 		return cmd.MarkFlagRequired(flag.HostID)
 	}
 }
+
+// Host returns the correct processName or the hostId in accordance with the service
```

Comment:
Happy to change the name of this function if you find a better one. I definitely need to read something about naming golang function 😅

Reply:
This name works, one what to read, Effective Go, is a great resource and here's a section on naming 
https://golang.org/doc/effective_go.html#names